src/MotionControl.c

```diff
@@ -94,6 +94,17 @@ Init_Trajectory_Status Ros_MotionControl_Init(rosidl_runtime_c__String__Sequence
         int  jointIndexInCtrlGroup;
         CtrlGroup* ctrlGroup;
 
+        //check to ensure there are no duplicate joint names in the list
+        for (checkForDupIndex = jointIndexInTraj; checkForDupIndex < sequenceGoalJointNames->size; checkForDupIndex += 1)
```

Comment:
It's late here, but wouldn't this lead to `checkForDupIndex == jointIndexInTraj` on the first iteration? At the very beginning of the `jointIndexInTraj` loop, that'd give `checkForDupIndex == jointIndexInTraj == 0`, which would compare the first joint name with itself?

Reply:
Yeah, that's supposed to be `+1`. Good catch.